irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/contract/service/EdcContractAgreementService.java

```diff
@@ -53,8 +54,11 @@ public List<ContractAgreement> getContractAgreements(final String... contractAgr
             throws ContractAgreementException {
 
         final QuerySpec querySpec = buildQuerySpec(contractAgreementIds);
+
+        final EndpointConfig endpoint = config.getControlplane().getEndpoint();
+        final String contractAgreements = endpoint.getContractAgreements();
         final ResponseEntity<EdcContractAgreementsResponse> edcContractAgreementListResponseEntity = edcRestTemplate.postForEntity(
-                config.getControlplane().getEndpoint().getContractAgreements() + EDC_REQUEST_SUFFIX, querySpec,
+                endpoint.getData() + contractAgreements + EDC_REQUEST_SUFFIX, querySpec,
```

Comment:
contractAgreements is used as a variable and data not why is this used mixed mode?

Reply:
see the lines above. endpoint is used to get the contractAgreements path. In order to avoid boilerplate code, the data can be extracted directly.